docs/testpmd.md

```diff
@@ -0,0 +1,250 @@
+
+## Pre-requisites
+
+This workload assumes a healthy cluster is runing with SR-IOV and Performance Add-On operators. 
+
+* [SR-IOV](https://docs.openshift.com/container-platform/4.6/networking/hardware_networks/installing-sriov-operator.html) operator is required to create/assign VFs to the pods
+* [PAO](https://docs.openshift.com/container-platform/4.6/scalability_and_performance/cnf-performance-addon-operator-for-low-latency-nodes.html) is required to prepare the worker node with Hugepages for DPDK application and isolate CPUs for workloads. 
+
+### Sample SR-IOV configuration
+
+These are used to create VFs and SRIOV networks, interface might vary
+
+```yaml
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: testpmd-policy
+  namespace: openshift-sriov-network-operator
+spec:
+  deviceType: vfio-pci
+  nicSelector:
+    pfNames:
+     - ens2f1
+  nodeSelector:
+    node-role.kubernetes.io/worker: ""
+  numVfs: 20
+  resourceName: intelnics
+```
+
+```yaml
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: testpmd-sriov-network
+  namespace: openshift-sriov-network-operator
+spec:
+  ipam: |
+    {
+      "type": "host-local",
+      "subnet": "10.57.1.0/24",
+      "rangeStart": "10.57.1.100",
+      "rangeEnd": "10.57.1.200",
+      "routes": [{
+        "dst": "0.0.0.0/0"
+      }],
+      "gateway": "10.57.1.1"
+    }
+  spoofChk: "on"
+  trust: "on"
+  resourceName: intelnics
+  networkNamespace: my-ripsaw
+
+```
+
+### Sample PAO configuration
+
+The CPU and socket information might vary, this is defined based on the recommendation [here](https://docs.google.com/document/d/1wlT8z-QNzibgmNZnw5M3QNbwBcE0bJEcjWwh_BoA1Ig/edit?usp=sharing)
+```yaml
+---
+apiVersion: performance.openshift.io/v1
+kind: PerformanceProfile
+metadata:
+   name: testpmd-performance-profile-0
+spec:
+   cpu:
+      isolated: "6,8,10,12,14,16,18,20,22,24,26,28"
+      reserved: "0-5,7,9,11,13,15,17,19,21,23,25,27,29"
+   hugepages:
+      defaultHugepagesSize: "1G"
+      pages:
+         - size: "1G"
+           count: 16
+           node: 0
+   numa:
+      topologyPolicy: "best-effort"
+   nodeSelector:
+      node-role.kubernetes.io/worker: ""
+   realTimeKernel:
+     enabled: false
+   additionalKernelArgs:
+   - nosmt
+   - tsc=reliable
+```
+
+## Running standalone TestPMD
+
+A minimum configuration to run testpmd with in a pod, 
+
+```yaml
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+kind: Benchmark
+metadata:
+  name: testpmd-benchmark
+  namespace: my-ripsaw
+spec:
+  clustername: myk8scluster
+  workload:
+    name: testpmd
+    args:
+      privileged: true
+      pin: false
+      pin_testpmd: "node-0"      
+      networks:
+        testpmd:
+          - name: testpmd-sriov-network
+            count: 2  # Interface count
+            mac:
+              - 60:60:00:f4:47:06
```

Comment:
Where does the user determine the MAC?

Reply:
These are optional, random mac input from user for the SR-IOV VF. If not given script generates it [here](https://github.com/cloud-bulldozer/benchmark-operator/pull/510/files#diff-8faf8cd1f3463c3da5f4e18c423fcb7e1922806af35cd54c5e8519b72a98c33aR7). User must keep track of these mac for traffic generation. 
Where as `peer_mac` is traffic generator interface mac.
I will update this in readme.